packages/api/src/routes/devices/internal-user.ts

```diff
@@ -0,0 +1,62 @@
+import { Response } from "express";
+import Router from "express-promise-router";
+import status from "http-status";
+import { asyncHandler } from "../util";
+import {
+  getAllConnectedUsers,
+  getConnectedUserOrFail,
+} from "../../command/connected-user/get-connected-user";
+import { Constants, providerOAuth2OptionsSchema } from "../../shared/constants";
+import { updateProviderData } from "../../command/connected-user/save-connected-user";
+import { sendProviderDisconnected } from "../../command/webhook/devices";
+
+const router = Router();
+
+/** ---------------------------------------------------------------------------
+ * POST /internal/user/refresh-tokens
+ *
+ * Attempts to refresh all users' tokens for each of their connected providers.
+ * If the refresh doesn't work, this means that the token is invalid and can no
+ * longer be used - in this case, the token will be revoked locally and a
+ * webhook will be sent to the CX to notify them of the disconnect.
+ *
+ *
+ * @return 200 OK
+ */
+router.post(
+  "/refresh-tokens",
+  asyncHandler(async (_, res: Response) => {
+    const connectedUsers = await getAllConnectedUsers();
```

Comment:
I suggest we add an optional `cxId` param on the request, and get connected users by customer, if possible.

Reasoning is that, as we move forward, hopefully we should get to a point where loading all connected users in memory at once will impact/break the API. Doing it by cx, with maybe a small sleep/wait between them should minimize this risk a lot.

Getting the list of cxs on OSS might be tricky (sadly), so maybe we could paginate the `getAllConnectedUsers`?

Another option could be to make it harder for someone to execute this w/o the `cxId` query param. I'm trying to avoid someone hitting this w/o much thought in 6 months time and breaking the App/platform - I expect we'll likely leave this untouched after its done. :)

Reply:
Sounds good WRT to pagination, but I think the `cxId` param is super simple to add and, should we need to execute this again, we can do it for a specific user: its easy to imagine a situation where, for whatever reason, we want to reset just for one cx. In such case, running this has the potential to impact (maybe negatively) all other cxs.